### ISSUE TITLE
Is "Processing x items" #679 issue back 

### ISSUE THREAD

**Issue by zell-mbc** [abraunegg/onedrive#744]
Since upgrading to onedrive v2.3.10-9-g0589113 yesterday, I am getting those constant notifications again, same as described in this case here: https://github.com/abraunegg/onedrive/issues/679
Anyone else seeing the same?

```
$ onedrive --version                                                  
onedrive v2.3.10-9-g0589113

```

**Comment by abraunegg** [comment]
@zell-mbc 
Please upgrade to the latest master and retry .. you are WAY WAY behind if running `v2.3.10-9`

**Comment by zell-mbc** [comment]
Oops, sorry. Clearly I shouldn't use the Arch AUR to retrieve my updates.
I switched to direct updates from here, problem solved!

`
$ onedrive --version

onedrive v2.3.12
`

**Comment by petobens** [comment]
I've recently upgrade to `v2.3.12` and now I'm getting those constant notifications. For the record here is my config:
```onedrive version                    = v2.3.12
Config path                         = /home/pedro/.config/onedrive
Config file found in config path    = true
Config option 'check_nosync'        = false
Config option 'sync_dir'            = /home/pedro/OneDrive
Config option 'skip_dir'            =
Config option 'skip_file'           = ~*|.~*|*.tmp
Config option 'skip_dotfiles'       = false
Config option 'skip_symlinks'       = false
Config option 'monitor_interval'    = 180
Config option 'min_notify_changes'  = 5
Config option 'log_dir'             = /var/log/onedrive/
Config option 'sync_root_files'     = false
Selective sync configured           = false
```
Is there a way to silence them? I'm happy to provide further debugging logs if needed (please let me know how though). Thanks in advance.

**Comment by abraunegg** [comment]
@petobens 
Please follow these directions to provide a debug log - https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support

You are not using sync_list, so this should not impact you.

**Comment by petobens** [comment]
Thanks @abraunegg for the quick reply. I've shared you the debug log file via Google Drive since it was too big to be sent in an email.

**Comment by abraunegg** [comment]
@petobens 
Thanks for the debug log.

What the log is showing me is that for 'whatever' reason, your local database was out of sync .. or .. your removing the file or using `--resync` at some point ...

I say this because after each of the 'Processing X changes' messages, for each processed item there is the following:
```
[DEBUG] OneDrive change is an update to an existing local item
[DEBUG] Updating local database with item details
[DEBUG] item details: Item(<redacted>)
```

This tells me that the DB locally is out of sync.

After processing all the object and all the processing messages, this then is reported in the log:
```
[DEBUG] OneDrive Reported no delta changes - Local path and OneDrive in-sync
```
and no more 'Processing X changes' messages are being sent to be displayed.

So in short .. I am not seeing an issue or problem at this stage, everything looks normal.

**Comment by petobens** [comment]
But this happens every single time after each reboot, that doesn't seem normal.... can I somehow make it go away?

**Comment by abraunegg** [comment]
@petobens 
> But this happens every single time after each reboot, that doesn't seem normal....

No .. that is not normal.

Give me ~30mins and I will have a PR for some extra debug details for you to run.

**Comment by abraunegg** [comment]
@petobens 
Please can you test the following PR, including your reboot scenario:
```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
git fetch origin pull/745/head:pr745
git checkout pr745
./configure; make clean; make;
```

As your 'rebooting' you will also need to 'install' this PR version.

**Comment by petobens** [comment]
I'm afk until tomorrow. Will try it as soon as I'm back. Thanks

**Comment by petobens** [comment]
I've sent over 3 files via Google Drive: the first one before the reboot. After rebooting I started onedrive with `onedrive --monitor --verbose --verbose > debug_output2_after_reboot.log 2>&1` and after 10 minutes kill that and started a new one with `onedrive --monitor --verbose --verbose > debug_output2_after_reboot2.log 2>&1` (in this second run the processing message appears once again). Mmm I generally run onedrive with systemd (and not manually as I did before): can systemd be related to my problem? Once again thanks for the patience and for looking into this.

**Comment by abraunegg** [comment]
@petobens 
I have looked over the logs and have the following comments:

1. When you compiled this version, you missed adding `--enable-notifications` .. so this version will not & cannot send notifications (`v2.3.12-3-g0ebef6d`) when your running it
2. Are you still getting the 'Processing X items' issue? It is unclear from above. If you installed the above version then you should not be getting any notification.

Looking at the log files specifically, this is now correctly being set:
```
syncListConfiguredOverride = false
```

Which means that the issue you were seeing when starting the application and performing a full sync validation occurring is not happening in any of your logs.

Can you please do the following:

1. Stop any running systemd service
2. Recompile from the PR source, but this time add  `--enable-notifications` 
3. Install this version
4. Restart any systemd process

**Comment by abraunegg** [comment]
@petobens
Any update from your side?

**Comment by Modelmat** [comment]
I have this issue too. Suggest reopening?

**Comment by abraunegg** [comment]
@Modelmat 
Please can you test the following PR as per above:
1. Stop any running onedrive systemd service

```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
git fetch origin pull/745/head:pr745
git checkout pr745
./configure --enable-notifications; make clean; make;
```
2. Restart onedrive systemd process

Once done, please advise.

**Comment by abraunegg** [comment]
@petobens
Any update from your side?

**Comment by Modelmat** [comment]
That seems to fix the generation of the message for me.

**Comment by abraunegg** [comment]
@Modelmat 
Please provide debug log by modifying your systemd files as per https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support and provide debug log via email

**Comment by petobens** [comment]
Sorry I'm tied at work these days. I'll report back as soon as I can. I apologize in advance.

**Comment by Modelmat** [comment]
@abraunegg I have a debug log but am I unsure of which email to send it to; `support@mynas.com.au` seems a weird email.

**Comment by abraunegg** [comment]
Nothing weird about it .. that is the right email address

**Comment by Modelmat** [comment]
Okay, will send.

**Comment by abraunegg** [comment]
@Modelmat 
email received.

In the debug log your still running v2.3.12 as per this:
```
Application version: v2.3.12
```

so you are not testing your issue against the PR version, which should be:
```
onedrive v2.3.12-3-g0ebef6d
```

**Comment by Modelmat** [comment]
I am, 
![image](https://user-images.githubusercontent.com/25718989/70684266-b0640200-1cf9-11ea-9f91-4740fd74af99.png)

**Comment by abraunegg** [comment]
@Modelmat 
No your not. In your log file, generated from your systemd process, as provided via email is v2.3.12 ... check your install path for your systemd script as to where the binary is running from and check that directly ...

**Comment by Modelmat** [comment]
I ran `onedrive --monitor` directly.

**Comment by abraunegg** [comment]
Please check your steps .. your not running the PR version.

Double check your steps, and use a single `--verbose` with your command line which will print you also the version your using when you run your command

**Comment by Modelmat** [comment]
I just did again. As far as I can tell, it does not have the commit number in the debug_output file, just the Application Version.
I ran `onedrive --version`, got `v2.3.12-3-g0ebef6d`, then ran 
`onedrive --monitor --verbose --verbose > debug_output.log 2>&1`

**Comment by abraunegg** [comment]
@Modelmat 
The logfile you emailed does not represent what you are saying.

Please can you do the following:
```
alex@alex-virtual-machine:~$ which onedrive
/usr/local/bin/onedrive
alex@alex-virtual-machine:~$ /usr/local/bin/onedrive --version
onedrive v2.3.12-3-g0ebef6d
alex@alex-virtual-machine:~$ cat /usr/lib/systemd/user/onedrive.service | grep ExecStart
ExecStart=/usr/local/bin/onedrive --monitor
alex@alex-virtual-machine:~$ /usr/local/bin/onedrive --version
onedrive v2.3.12-3-g0ebef6d
alex@alex-virtual-machine:~$ onedrive --version
onedrive v2.3.12-3-g0ebef6d
alex@alex-virtual-machine:~$
```

When I start / restart the systemd service (if I have modified it to be verbose- which I have) I get the following:
```
...
Dec 12 18:03:52 alex-virtual-machine onedrive[4353]: All operations will be performed in: /home/alex/OneDrive
Dec 12 18:03:55 alex-virtual-machine onedrive[4353]: Application version: v2.3.12-3-g0ebef6d
Dec 12 18:03:55 alex-virtual-machine onedrive[4353]: Account Type: personal
Dec 12 18:03:55 alex-virtual-machine onedrive[4353]: Default Drive ID: 66d53be8a5056eca
Dec 12 18:03:55 alex-virtual-machine onedrive[4353]: Default Root ID: 66D53BE8A5056ECA!101
Dec 12 18:03:55 alex-virtual-machine onedrive[4353]: Remaining Free Space: 5064162652
...
```

Note that the PR version is listed.

If I run 'manually':
```
alex@alex-virtual-machine:~$ 
alex@alex-virtual-machine:~$ onedrive --monitor --verbose
Using Config Dir: /home/alex/.config/onedrive
Using logfile dir: /var/log/onedrive/
Notification (dbus) server not available, disabling
Initializing the OneDrive API ...
Opening the item database ...
All operations will be performed in: /home/alex/OneDrive
Application version: v2.3.12-3-g0ebef6d
Account Type: personal
Default Drive ID: 66d53be8a5056eca
Default Root ID: 66D53BE8A5056ECA!101
...
```

Again the PR version is listed.

Please can you double check your process to ensure you have installed the PR version, restarted systemd correctly, and validating that your notifications are coming through correctly without 'Processing X items' constantly.

If you think your still having an issue, please generate a debug log as per the directions and email this over, as the debug log you have provided is for the version that is older than the PR version.

**Comment by Modelmat** [comment]
I am still getting all of "processing" with the PR.
```
❯ which onedrive
/usr/local/bin/onedrive
                                                                                                                                                                          0.05 19:46:50
❯ /usr/local/bin/onedrive --version
onedrive v2.3.12-3-g0ebef6d
                                                                                                                                                                          0.20 19:46:57
❯ onedrive --version
onedrive v2.3.12-3-g0ebef6d
```
I still get `Application version: v2.3.12`.
`make uninstall` removes the `/usr/local/bin/onedrive`, and redoing `make install` adds the file again - this is the one from git.

Furthermore, 
```
❯ git status                          
On branch pr745
nothing to commit, working tree clean
```

**Comment by abraunegg** [comment]
@Modelmat 
> I still get Application version: v2.3.12.

You have the binary installed somewhere in a search path somewhere on your system or you are not stopping all your onedrive instances - and one is still running.

Suggest:
1. `make uninstall`
2. `sudo find / | grep 'onedrive'`

Work out where the rogue binary is installed.

When you are looking at your logs or running the application it will display the full version.

The only way that it is not doing this, is that the binary your running is v2.3.12 - there is zero other way for the application to be built as the PR and display a different version as to when you pass `--version` to the binary you think your running.

**Comment by abraunegg** [comment]
@Modelmat 
And you also did not provide the output of your systemd file & the file location that it is trying to run:
```
alex@alex-virtual-machine:~$ which onedrive
/usr/local/bin/onedrive
alex@alex-virtual-machine:~$ /usr/local/bin/onedrive --version
onedrive v2.3.12-3-g0ebef6d
alex@alex-virtual-machine:~$ cat /usr/lib/systemd/user/onedrive.service | grep ExecStart
ExecStart=/usr/local/bin/onedrive --monitor
alex@alex-virtual-machine:~$ /usr/local/bin/onedrive --version
onedrive v2.3.12-3-g0ebef6d
alex@alex-virtual-machine:~$ onedrive --version
onedrive v2.3.12-3-g0ebef6d
alex@alex-virtual-machine:~$
```

**Comment by Modelmat** [comment]
I'm not running it with systemd, so I'm not sure why it's an issue.

**Comment by abraunegg** [comment]
@Modelmat 
> I'm not running it with systemd ( for testing ), so I'm not sure why it's an issue.

Because you have then 2 version of the file on your system and your logs etc will be picking up the installed version - this is why this is an issue .... and potentially why your not seeing this issue as fixed.

**Comment by Modelmat** [comment]
Oh.
(As a side note, why would it matter to have two installed, this seems bad?)
I just uninstalled the other version and now it's showing the version from git.
Sorry about that.
I'm going to remove some personal info and send it again.

**Comment by abraunegg** [comment]
@Modelmat 
> I just uninstalled the other version and now it's showing the version from git.

Did you 'stop' all your systemd services first?

If not - the old binary is still executing and running in memory ... 

Make sure you stop all systemd processes, then uninstall, confirm that the old binary is removed, then install the PR version, then re-enable / start the systemd service.

**Comment by Modelmat** [comment]
My systemd service has been stopped for ages. It was not running.
Removing the old binary removes the systemd service and I am running the new binary manually.

**Comment by abraunegg** [comment]
@Modelmat 
> Removing the old binary removes the systemd service and I am running the new binary manually.

Whilst true, does not 'stop' the existing process if it is still running.

Please follow these directions to provide a debug log if the issue is persisting - https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support

**Comment by Modelmat** [comment]
It is stopped. It has been stopped.
```
❯ systemctl status onedrive.service     
Unit onedrive.service could not be found.

~                                                                                                                                                                              4 0.29 11:15:30
❯ killall onedrive
onedrive: no process found
```
I have emailed you the debug log running with the latest version of the application, with the Application version specifying the correct one you wanted.

I have no idea why you insist that I am not running the correct version when I have already shown you I am and it is still not working.

**Comment by abraunegg** [comment]
@Modelmat 
The logfile you provided on Thu 12/12/2019 4:00 PM is the only logfile you have ever provide for this issue that you say you have. In that logfile, you are running the v2.3.12 release, not the PR version as requested. As such, that logfile is invalid.

Since then, I have been asking you to ensure your running the right version, which you now say you are running - great. You also say you still have the issue ... OK .. no problem. Please follow these directions to provide a debug log that is captured when the issue is occuring - https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support

Without the debug log, I cannot determine what you are seeing, or what is causing the issue for you.

Constantly responding that it is not working but without evidence is hard for me to go inspect and work out what is wrong is where.

**Comment by Modelmat** [comment]
Sorry, I had sent an email. It must have been lost in the depths of the internet. I have resent the debug log now.

**Comment by abraunegg** [comment]
@Modelmat 
Thanks for the updated log file.

Based on that logfile, please can you rebuild your client from the PR which has been updated. When running the PR version you should now have `onedrive v2.3.12-7-g28a1aaa`

Please can you re-test against the issue you are seeing.

**Comment by abraunegg** [comment]
@petobens , @Modelmat 
Any update from your side ?

**Comment by Modelmat** [comment]
Sorry, I've been a bit sick.
I'll see if I can test it tomorrow morning if I'm feeling better.

**Comment by Modelmat** [comment]
Okay, I tested it and I haven't received any notifications yet. Been running for about 5 minutes. Thanks :)

**Comment by abraunegg** [comment]
@petobens 
Any update from your side?

**Comment by abraunegg** [comment]
@petobens, @Modelmat 
FYI the PR is also now merged into 'master' - when running master with this fix, the version will be `onedrive v2.3.12-5-g217d724`

**Comment by petobens** [comment]
Thanks! Sorry for the lack of feedback. Will try it out.

**Comment by Modelmat** [comment]
Thanks. Any ETA on the next release?

**Comment by abraunegg** [comment]
@Modelmat 
Releasea are typically monthly at the start of the month or as needed if high severity bug / crash. Next release will not be until early January.

**Comment by Modelmat** [comment]
Okay, that's fine.

**Comment by abraunegg** [comment]
@zell-mbc 
> Thought I'd share some more confirmation, my install started to produce these messages again as well, even on the latest offiicial version. Resync wouldn't help.

Which version was this? v2.3.12 ? If yes, then is was fixed via https://github.com/abraunegg/onedrive/commit/217d72486d55d313d9fda727fc5e2b06c0900de6

> I just now installed the patched version and things are back to normal again!

Which version was this? Current 'master' should be v2.3.12-5 ?

**Comment by zell-mbc** [comment]
Right now I am on onedrive v2.3.12-11-g59cfa6c
Before I was on onedrive v2.3.12, not sure about the -5 

In any case, what made me open this thread was that I saw the "processing" meassages re-appear all of a sudden which was cleary due to me installing an out of date version by accident.
Based on your response higher up I installed 2.3.12 and, after some settling down, the messages seemed to have disapeared. 
But then, probably after each reboot the messages started again, only settling down after a couple of hours. But during that time flooding my journal log.

Now with v2.3.12-11-g59cfa6c I don't see any of these "processing" messages any longer.All clear after 2 days, with reboots.

**Comment by abraunegg** [comment]
@zell-mbc 
> Now with v2.3.12-11-g59cfa6c I don't see any of these "processing" messages any longer.All clear after 2 days, with reboots.

Unsure where / what your building from, as this is the current 'master' that you should be using:

```
[alex@centos7full code]$ ./clone_master.sh 
Cloning into 'onedrive'...
remote: Enumerating objects: 36, done.
remote: Counting objects: 100% (36/36), done.
remote: Compressing objects: 100% (36/36), done.
remote: Total 3697 (delta 19), reused 0 (delta 0), pack-reused 3661
Receiving objects: 100% (3697/3697), 2.25 MiB | 1.04 MiB/s, done.
Resolving deltas: 100% (2534/2534), done.
checking for a BSD-compatible install... /usr/bin/install -c
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for dmd... dmd
checking version of D compiler... 2.085.0
checking for curl... yes
checking for sqlite... yes
checking for notify... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating contrib/pacman/PKGBUILD
config.status: creating contrib/spec/onedrive.spec
config.status: creating onedrive.1
config.status: creating contrib/systemd/onedrive.service
config.status: creating contrib/systemd/onedrive@.service
rm -f onedrive onedrive.o version
rm -rf autom4te.cache
rm -f config.log config.status
if [ -f .git/HEAD ] ; then \
        git describe --tags > version ; \
else \
        echo v2.3.13-dev > version ; \
fi
dmd  -w -g -O -J. -version=NoPragma -version=NoGdk -version=Notifications -L-lcurl -L-lsqlite3 -L-lnotify -L-lgdk_pixbuf-2.0 -L-lgio-2.0 -L-lgobject-2.0 -L-lglib-2.0 -L-ldl src/config.d src/itemdb.d src/log.d src/main.d src/monitor.d src/onedrive.d src/qxor.d src/selective.d src/sqlite.d src/sync.d src/upload.d src/util.d src/progress.d src/notifications/notify.d src/notifications/dnotify.d -ofonedrive
[alex@centos7full code]$
[alex@centos7full code]$ cd onedrive
[alex@centos7full onedrive]$ ./onedrive --version
onedrive v2.3.12-5-g217d724
[alex@centos7full onedrive]$
```
